dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeMetadataConnector.java

```diff
@@ -48,6 +50,8 @@ public class SnowflakeMetadataConnector extends AbstractSnowflakeConnector
   @SuppressWarnings("UnusedVariable")
   private static final Logger LOG = LoggerFactory.getLogger(SnowflakeMetadataConnector.class);
 
+  private static final Set<Class<?>> validatedTables = new HashSet<>();
```

Comment:
At first review, this looks like a static mutable variable, which makes the dumper itself not be reusable in a thread-safe scenario.